cmdstanpy/cmdstan_args.py

```diff
@@ -0,0 +1,398 @@
+            if not os.path.exists(os.path.dirname(self.output_basename)):
+                raise ValueError(
+                    'invalid path for output files: {}'.format(
+                        self.output_basename)
+                )
+            try:
+                with open(self.output_basename, 'w+') as fd:
+                    pass
+                os.remove(self.output_basename)  # cleanup
+            except Exception:
+                raise ValueError(
+                    'invalid path for output files: {}'.format(
+                        self.output_basename)
+                )
+            if self.output_basename.endswith('.csv'):
+                self.output_basename = self.output_basename[:-4]
+
+        if self.seed is None:
+            rng = np.random.RandomState()
+            self.seed = rng.randint(1, 99999 + 1)
+        else:
+            if not isinstance(self.seed, (int, list)):
+                raise ValueError(
+                    'seed must be an integer between 0 and 2**32-1,'
+                    ' found {}'.format(self.seed)
+                )
+            elif isinstance(self.seed, int):
+                if self.seed < 0 or self.seed > 2 ** 32 - 1:
+                    raise ValueError(
+                        'seed must be an integer between 0 and 2**32-1,'
+                        ' found {}'.format(self.seed)
+                    )
+            else:
+                if self.chain_ids is None:
+                    raise ValueError(
+                        "seed must not be a list when no chains used"
+                    )
+
+                if len(self.seed) != len(self.chain_ids):
+                    raise ValueError(
+                        'number of seeds must match number of chains '
+                        ' found {} seed for {} chains '.format(
+                            len(self.seed), len(self.chain_ids)
+                        )
+                    )
+                for i in range(len(self.seed)):
+                    if self.seed[i] < 0 or self.seed[i] > 2 ** 32 - 1:
+                        raise ValueError(
+                            'seed must be an integer value'
+                            ' between 0 and 2**32-1,'
+                            ' found {}'.format(self.seed[i])
+                        )
+
+        if self.data is not None:
+            if not os.path.exists(self.data):
+                raise ValueError('no such file {}'.format(self.data))
+
+        if self.inits is not None:
+            if isinstance(self.inits, (int, float)):
+                if self.inits < 0:
+                    raise ValueError(
+                        'inits must be > 0, found {}'.format(self.inits)
+                    )
+            elif isinstance(self.inits, str):
+                if not os.path.exists(self.inits):
+                    raise ValueError('no such file {}'.format(self.inits))
+            elif isinstance(self.inits, List):
```

Comment:
list or (list, tuple) or List or (List, Tuple)?

Reply:
PEP8 says names of built-in types are lowercase -
https://www.python.org/dev/peps/pep-0008/#class-names